Export `GotRequestMethod` TypeScript type (#1017)
sindresorhus/got: readme.md

A read-only boolean describing whether the defaults are mutable or not. If set to `true`, you can [update headers over time](#hooksafterresponse), for example, update an access token when it expires.

## Types

When used in a TypeScript project, Got has a set of handy types and interfaces available.

### Got

> Typescript will automatically infer types for Got instances but in case you want to define something like dependencies you can use following: `Got`, `GotStream`, `ReturnStream`, `GotRequestMethod`, `GotReturn`

Reviewer: I'm not sure the value of just listing these types. The user has no idea what they do or what to use them for. Maybe better to just mention one type with example and refer to the type definition instead. It's also annoying to have to keep this in sync with the source.

Same with hooks.

Author: I was thinking exactly the same when I was writing it, but I had to start somewhere. I'll try to shorten things up

A read-only boolean describing whether the defaults are mutable or not. If set to `true`, you can [update headers over time](#hooksafterresponse), for example, update an access token when it expires.

## Types

Since Got was rewritten in TypeScript it has a set of handy types and interfaces available

### Got
Typescript will automatically infer types for Got instances but in case you want to define something like dependencies you can use following: `Got`, `GotStream`, `ReturnStream`, `GotRequestMethod`, `GotReturn`

```typescript
import { GotRequestMethod } from 'got'

interface Dependencies {
  readonly post: GotRequestMethod
}
```

### Hooks
Following types can be used to define isolated hooks and keep their interfaces consistent: `Hooks`, `HookEvent`, `HookType`, `InitHook`, `BeforeRequestHook`, `BeforeRedirectHook`, `BeforeRetryHook`, `BeforeErrorHook`, `AfterResponseHook`

```typescript
const addAccessToken = (accessToken: string): BeforeRequestHook => (options) => {
  options.path = `${options.path}?access_token=${accessToken}`
}
```

>  ## Errors

Reviewer 2: There's an unnecessary space before `## Errors`

Author: fixed

## Types

> When used in a TypeScript project, Got has a set of handy types and interfaces available.

Reviewer 2: ```suggestion
Got has a set of handy types and interfaces available, so you can integrate Got with your TypeScript project seamlessly.
```

### Got

> Typescript will automatically infer types for Got instances but in case you want to define something like dependencies you can import available types directly from `got`

Reviewer 2: Missing dot at the end

Author: Fixed
